File: parse_pdf.py
import PyPDF2
import sys

def extract_text(pdf_path):
    with open(pdf_path, 'rb') as file:
        reader = PyPDF2.PdfReader(file)
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        
        with open('extracted_pdf.txt', 'w', encoding='utf-8') as out_file:
            out_file.write(text)

if __name__ == '__main__':
    extract_text('Daily Bus Routes From 16.02.2026 (1).pdf')

File: parse_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

let dataBuffer = fs.readFileSync('Daily Bus Routes From 16.02.2026 (1).pdf');

pdf(dataBuffer).then(function (data) {
    fs.writeFileSync('extracted_pdf.txt', data.text);
    console.log('PDF extracted successfully to extracted_pdf.txt');
}).catch(function (error) {
    console.error(error);
});
